lms/services/upsert.py

```diff
@@ -5,6 +5,20 @@
 from zope.sqlalchemy import mark_changed
 
 
+def upsert(db, model, index_columns: list[str], update_columns: list[str]):
```

Comment:
Single element upsert in SQLA core.

We could go this route or just write the conditional (if doesn't exists create, else, update) in python.